src/components/PopupWithConfirmation.js

```diff
@@ -7,14 +8,27 @@ export default class PopupWithConfirmation extends Popup {
   }
 
   setSubmitAction(submitAction) {
```

Comment:
You don't need `setSubmitAction` anymore because you update the card and its id.